DQM/SiStripMonitorCluster/src/SiStripMonitorCluster.cc

```diff
@@ -732,227 +825,232 @@ void SiStripMonitorCluster::analyze(const edm::Event& iEvent, const edm::EventSe
+          if(ampls[iamp]>0){ // nonzero amplitude
+            cluster_signal += ampls[iamp];
+            if(!qualityHandle->IsStripBad(qualityRange, clusterIter->firstStrip()+iamp)){
+              noise = noiseHandle->getNoise(clusterIter->firstStrip()+iamp,detNoiseRange)/gainHandle->getStripGain(clusterIter->firstStrip()+iamp, detGainRange);
+            }
+            noise2 += noise*noise;
+            nrnonzeroamplitudes++;
+          }
+        } // End loop over cluster amplitude
+
+        if (nrnonzeroamplitudes > 0) cluster_noise = sqrt(noise2/nrnonzeroamplitudes);
+
+        if( applyClusterQuality_ &&
+          (cluster_signal/cluster_noise < sToNLowerLimit_ ||
+            cluster_signal/cluster_noise > sToNUpperLimit_ ||
+            cluster_width < widthLowerLimit_ ||
+            cluster_width > widthUpperLimit_) ) continue;
+
+            ClusterProperties cluster_properties;
+            cluster_properties.charge    = cluster_signal;
+            cluster_properties.position  = cluster_position;
+            cluster_properties.start     = cluster_start;
+            cluster_properties.width     = cluster_width;
+            cluster_properties.noise     = cluster_noise;
+
+            // Fill Module Level MEs
+            if (found_module_me) fillModuleMEs(mod_single, cluster_properties);
+
+            // Fill Layer Level MEs
+            if (found_layer_me) {
+              fillLayerMEs(layer_single, cluster_properties);
+              if (layerswitchclusterwidthprofon)
+              layer_single.LayerClusterWidthProfile->Fill(iDet, cluster_width);
+            }
+
+            if (subdetswitchcluschargeon || subdetswitchcluswidthon)
+            {
+              std::map<std::string, SubDetMEs>::iterator iSubdet  = SubDetMEsMap.find(subdet_label);
+              if(iSubdet != SubDetMEsMap.end())
+              {
+                if (subdetswitchcluschargeon) iSubdet->second.SubDetClusterChargeTH1->Fill(cluster_signal);
+                if (subdetswitchcluswidthon)  iSubdet->second.SubDetClusterWidthTH1->Fill(cluster_width);
+              }
+            }
+
+
+            if (subdet_clusterWidth_vs_amplitude_on){
+              std::map<std::string, SubDetMEs>::iterator iSubdet  = SubDetMEsMap.find(subdet_label);
+              if(iSubdet != SubDetMEsMap.end()) iSubdet->second.SubDetClusWidthVsAmpTH2->Fill(cluster_signal, cluster_width);
+            }
+
+            if ( clusterWidth_vs_amplitude_on ) {
+              ClusWidthVsAmpTH2->Fill(cluster_signal, cluster_width);
+            }
+
+            if (subdetswitchtotclusprofon) {
+              std::map<std::string, SubDetMEs>::iterator iSubdet  = SubDetMEsMap.find(subdet_label);
+              std::pair<std::string,int32_t> det_layer_pair = folder_organizer.GetSubDetAndLayer(detid, tTopo);
+              if(iSubdet != SubDetMEsMap.end()) iSubdet->second.SubDetNumberOfClusterPerLayerTrend->Fill(trendVar,abs(det_layer_pair.second));
```

Comment:
Something odd happening with the file change history here as this is not code implemented by me.